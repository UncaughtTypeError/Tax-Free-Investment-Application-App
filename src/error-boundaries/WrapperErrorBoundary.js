import React from 'react';
// Theme
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

const useStyles = (theme) => ({
    typography: {
        fontWeight: 300,
        marginBottom: theme.spacing(5),
    }
});

class WrapperErrorBoundary extends React.Component {
    constructor(props) {
        super(props);
        this.state = { hasError: false };
    }

    static getDerivedStateFromError(error) {
        return { hasError: true };
    }

    componentDidCatch(error, errorInfo) {
        console.error({error}, errorInfo);
    }

    render() {

    const classes = this.props;

    if (this.state.hasError) {
        return (
            <Typography className={classes.typography} variant="h5" align="center">Sorry, something went wrong :(</Typography>
        );
    }

        return this.props.children; 
    }
}

export default withStyles(useStyles)(WrapperErrorBoundary);