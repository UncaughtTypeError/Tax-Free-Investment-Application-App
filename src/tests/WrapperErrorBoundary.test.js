import React from 'react';
// Enzyme
import { shallow } from 'enzyme';
// Component
import WrapperErrorBoundary from '../error-boundaries/WrapperErrorBoundary';

it('should render fallback content', () => {
    expect(shallow(<WrapperErrorBoundary>
        {() => { throw new Error(); }}
    </WrapperErrorBoundary>)).toMatchSnapshot();
});