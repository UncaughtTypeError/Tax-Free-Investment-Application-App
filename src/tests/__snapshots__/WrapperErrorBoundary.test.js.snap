// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render fallback content 1`] = `ShallowWrapper {}`;
